src/tdastro/math_nodes/basic_math_node.py

```diff
@@ -125,7 +128,16 @@ def __init__(self, expression, backend="numpy", node_label=None, **kwargs):
         # Create a function from the expression. Note the expression has
         # already been sanitized and validated via _prepare().
         def eval_func(**kwargs):
-            return eval(self.expression, globals(), kwargs)
+            try:
+                return eval(self.expression, globals(), kwargs)
+            except Exception as problem:
+                # Provide more detailed logging, including the expression and parameters
+                # used, when we encounter a math error like divide by zero.
+                logger.error(
+                    f"{type(problem)} encountered during operation: {self.expression}\n"
+                    f"with arguments={kwargs}"
+                )
+                raise problem
```

Comment:
Could we add some context to an exception and raise with a better message and/or traceback?